tests/RedisMutexTest.php

```diff
@@ -2,168 +2,144 @@
-        $this->assertNull($connection->get($lockKeyPrefix . $testLockName));
+        // Double release
+        $this->assertFalse($mutex->release(static::$mutexName));
+        $this->assertMutexKeyNotInRedis();
     }
 
-    public function testConcurrentAcquireLock()
+    public function testExpiration()
     {
-        $testLockName = 'the_lock';
+        $mutex = $this->createMutex();
 
-        $connection = $this->getConnection(true);
+        $this->assertTrue($mutex->acquire(static::$mutexName));
+        $this->assertMutexKeyInRedis();
+        $this->assertLessThanOrEqual(1500, $mutex->redis->executeCommand('PTTL', [$this->getKey(static::$mutexName)]));
 
-        $mutex1 = $this->getMutexInstance($connection);
-        $mutex2 = $this->getMutexInstance($connection);
+        sleep(2);
 
-        $mutex1->acquire($testLockName);
-        $this->assertFalse($mutex2->acquire($testLockName));
-
-        $mutex1->release($testLockName);
-        $this->assertTrue($mutex2->acquire($testLockName));
+        $this->assertMutexKeyNotInRedis();
+        $this->assertFalse($mutex->release(static::$mutexName));
+        $this->assertMutexKeyNotInRedis();
     }
 
-    public function testConcurrentReleaseLock()
+    public function acquireTimeoutProvider()
     {
-        $testLockName = 'the_lock';
-
-        $connection = $this->getConnection(true);
-
-        $mutex1 = $this->getMutexInstance($connection);
-        $mutex2 = $this->getMutexInstance($connection);
-
-        $mutex1->acquire($testLockName);
-        $this->assertFalse($mutex2->release($testLockName));
+        return [
+            'no timeout (lock is held)' => [0, false, false],
+            '2s (lock is held)' => [1, false, false],
+            '3s (lock will be auto released in acquire())' => [2, true, false],
+            '3s (lock is auto released)' => [2, true, true],
+        ];
     }
 
-    public function testLockTimeoutSuccess()
+    /**
+     * @covers \yii\redis\Mutex::acquireLock
+     * @covers \yii\redis\Mutex::releaseLock
+     * @dataProvider acquireTimeoutProvider
+     */
+    public function testConcurentMutexAcquireAndRelease($timeout, $canAcquireAfterTimeout, $lockIsReleased)
     {
-        $testLockName = 'the_lock';
-
-        $connection = $this->getConnection(true);
-
-        $mutex1 = $this->getMutexInstance($connection, [
-            'lockExpire' => 100,
-        ]);
-        $mutex2 = $this->getMutexInstance($connection);
+        $mutexOne = $this->createMutex();
+        $mutexTwo = $this->createMutex();
+
+        $this->assertTrue($mutexOne->acquire(static::$mutexName));
+        $this->assertFalse($mutexTwo->acquire(static::$mutexName));
+        $this->assertTrue($mutexOne->release(static::$mutexName));
+        $this->assertTrue($mutexTwo->acquire(static::$mutexName));
+
+        if ($canAcquireAfterTimeout) {
+            // Mutex 2 auto released the lock or it will be auto released automatically
```

Comment:
thanks for refactoring the tests. Looks a lot more readable and understandable now.
With respect to this `if` statement. This is a sign that your unit test is still doing too much and shows room for further refactoring in more self contained tests.